framework/packages/abstract-interface/src/account/proxy.rs

```diff
@@ -1,16 +1,19 @@
-use crate::Manager;
 pub use abstract_core::proxy::{ExecuteMsgFns as ProxyExecFns, QueryMsgFns as ProxyQueryFns};
-use abstract_core::{
-    objects::{price_source::UncheckedPriceSource, AssetEntry},
-    proxy::*,
-    MANAGER, PROXY,
-};
+use abstract_core::{objects::AccountId, proxy::*, PROXY};
 
 use cw_orch::{interface, prelude::*};
 
 #[interface(InstantiateMsg, ExecuteMsg, QueryMsg, MigrateMsg)]
 pub struct Proxy<Chain>;
 
+// TODO: would be nice to hide default `new`
+impl<Chain: CwEnv> Proxy<Chain> {
+    pub fn new_from_id(account_id: &AccountId, chain: Chain) -> Self {
```

Comment:
I'm wondering if the presumption is going to be that this account already has its state stored locally? So when you create this object and you interact with it you're not sure it has the state it requires (its address) as that's only available in the version control. 

In that case you need to load it from versions control directly or?